tests/resolve/test_pex_repository_resolver.py

```diff
@@ -119,6 +120,13 @@ def pex_repository(py27, py310, foreign_platform, manylinux):
     )
 
 
+@pytest.mark.skipif(
```

Comment:
Hrm, it looks like these 2 Python 3.5 failures are not bitrot. Main is definitely green for them. I'll circle back in the am and figure these two out before proceeding.

Reply:
Ok, yeah - this caught a bug. When I eliminated `iter_platform_args()`, that killed sending Pip the `--platform ... --implementation ... --python-version ... --abi ...` quad. The foreign_platform patches cover `--platform ... --implementation ... --abi ...`, aka: wheel tag matching, but they do not cover `--python-version ...`, aka `Requires-Python` matching. Pip requires you send the quad as a whole; so the `--python-version ...` / `Requires-Python` matching must be patched too. That patching is already implemented in the locker patches; so I'll need to share that code here for foreign platform resolves.

Reply:
Ok, I've reverted the test skips and updated the PR to include patching `Requires-Python` for foreign platform resolves. The PR update is large, but its mainly shuffling. The code to patch `Requires-Python` already existed to support universal lock resolves and so 95% of the delta here is extracting that bit of patching and re-arranging for re-use in the foreign platform patches.